src/main/java/it/gov/pagopa/reporting/client/ApiConfigClient.java

```diff
@@ -0,0 +1,134 @@
+            System.getenv("RANDOMIZATION_FACTOR") != null ? Double.parseDouble(System.getenv("RANDOMIZATION_FACTOR")) : 0.5;
+
+    public static ApiConfigClient getInstance() {
+        if (instance == null) {
+            instance = new ApiConfigClient();
+        }
+        return instance;
+    }
+
+    public CacheResponse getCache() throws IOException, IllegalArgumentException, Cache5XXException, Cache4XXException {
+        GenericUrl url = new GenericUrl(apiConfigCacheHost + getCacheDetails);
+        HttpRequest request = this.buildGetRequestToApiConfigCache(url);
+
+        if (enableRetry) {
+            this.setRequestRetry(request);
+        }
+
+        return this.executeCallToApiConfigCache(request);
+    }
+
+    public HttpRequest buildGetRequestToApiConfigCache(GenericUrl url) throws IOException {
+
+        HttpRequestFactory requestFactory = httpTransport.createRequestFactory(
+                (HttpRequest request) ->
+                        request.setParser(new JsonObjectParser(jsonFactory))
+        );
+
+        HttpRequest request = requestFactory.buildGetRequest(url);
+        HttpHeaders headers = request.getHeaders();
+        headers.set("Ocp-Apim-Subscription-Key", apiKey);
+        return request;
+    }
+
+    public void setRequestRetry(HttpRequest request) {
+        /**
+         * Retry section config
+         */
+        ExponentialBackOff backoff = new ExponentialBackOff.Builder()
+                .setInitialIntervalMillis(initialIntervalMillis)
+                .setMaxElapsedTimeMillis(maxElapsedTimeMillis)
+                .setMaxIntervalMillis(maxIntervalMillis)
+                .setMultiplier(multiplier)
+                .setRandomizationFactor(randomizationFactor)
+                .build();
+
+        // Exponential Backoff is turned off by default in HttpRequest -> it's necessary include an instance of HttpUnsuccessfulResponseHandler to the HttpRequest to activate it
+        // The default back-off on anabnormal HTTP response is BackOffRequired.ON_SERVER_ERROR (5xx)
+        request.setUnsuccessfulResponseHandler(
+                new HttpBackOffUnsuccessfulResponseHandler(backoff));
+    }
+
+    public CacheResponse executeCallToApiConfigCache(HttpRequest request) throws IOException, IllegalArgumentException, Cache5XXException, Cache4XXException {
+
+        Type type = new TypeToken<List<CreditorInstitutionStation>>() {}.getType();
+
+        ObjectMapper mapper = new ObjectMapper();
+        CacheResponse cacheResponse = CacheResponse.builder().build();
+        List<CreditorInstitutionStation> creditorInstitutionStationList = new ArrayList<>();
+        List<Station> stationList = new ArrayList<>();
+        try {
+            InputStream resIs = request.execute().getContent();
+            Map<String,Object> responseMap = mapper.readValue(resIs, HashMap.class);
+            Map<String,Object> creditorInstitutionStations = (HashMap) responseMap.get("creditorInstitutionStations");
```

Comment:
this cast to `HashMap` is necessary?

Reply:
Yes, otherwise the mapping of the structure will not work properly